CONTRIBUTING.md

```diff
@@ -77,20 +85,272 @@ the [Corporate Contributor License Agreement](/docs/SAP%20Corporate%20Contributo
+
+### Type
+Must be one of the following:
+
+* **build**: Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Circle, BrowserStack, SauceLabs)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix
+* **refactor**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* **test**: Adding missing tests or correcting existing tests
+
+### Scope
+The scope should be the name of the npm package affected (as perceived by the person reading the changelog generated from commit messages.
+
+The following is the list of supported scopes:
+
+* **core**
+* **platform**
+* **doc-app**
+
+There are currently a few exceptions to the "use package name" rule:
+
+* **changelog**: used for updating the release notes in CHANGELOG.md
+* **doc-app**: used for docs-app (angular.io) related changes within the /aio directory of the
+  repo
+* none/empty string: useful for `style`, `test` and `refactor` changes that are done across all
+  packages (e.g. `style: add missing semicolons`) and for docs changes that are not related to a
+  specific package (e.g. `docs: fix typo in tutorial`).
+
+### Subject
+The subject contains a succinct description of the change:
+
+* use the imperative, present tense: "change" not "changed" nor "changes"
+* don't capitalize the first letter
+* no dot (.) at the end
+
+### Body
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+The body should include the motivation for the change and contrast this with previous behavior.
+
+### Footer
+The footer should contain any information about **Breaking Changes** and is also the place to
+reference GitHub issues that this commit **Closes**.
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.
+
+
+
+## <a name="pr"></a> Create PR Guidelines
+
+if you look at the format of our npm packages you will see following:
+
+`0.11.0-rc.10` where you can read this as `MAJOR` . `MINOR` . `PATCH PR-BUILD`.
+  
+and depending on the PR title we increment one of these numbers. 
+  * This is happening as part of the CI process that reads your **PR Title (not commit message)** and decide which number to increment. 
+
+### PR Title Format
+
+The PR title must follow this format:
+
+
+
+```
+<type>: <subject>
+```
+
+Samples:
+
+```
+feat: My adding new binding to button component
+```
+
+
+```
+chore: Updated package.json
+```
+
+
+
+The PR check process fails if you put something else then `WIP|feat|chore|test|doc|fix`
```

Comment:
WIP won't accept it for merging

Reply:
Should we mention this in the docs?

Reply:
updated text it make it more sense. that `feat|chore|test|doc|fix` are accepted for merge, where WIP is also valid title but its only really for work in progress and it will not allow merging, its really used to create a PR, run a few tests on CI, let it reviewed and signal to others this is what I am working on..